File: JavaScript/1143-Longest Common Subsequence.js
/**
 * @param {string} text1
 * @param {string} text2
 * @return {number}
 */
var longestCommonSubsequence = function(text1, text2){
    let len1 = text1.length, len2 = text2.length;
    let dp = new Array(len2+1).fill().map(()=>new Array(len1+1).fill(0));

    for (let r=1; r<=len2; r++){
        for (let c=1; c<=len1; c++){
            dp[r][c] = text2[r-1] === text1[c-1]? dp[r-1][c-1] + 1 : Math.max(dp[r][c-1], dp[r-1][c]);
        }
    }
    
    return dp[len2][len1];
};

File: C++/1143-Longest Common Subsequence.cpp
class Solution {
public:
    int longestCommonSubsequence(string text1, string text2) {
        int N1 = text1.size(), N2 = text2.size();
        vector<vector<int>> dp(N1+1, vector<int>(N2+1, 0));
        
        for (int i=1; i<N1+1; i++) {
            for (int j=1; j<N2+1; j++) {
                if (text1[i-1] == text2[j-1]) {
                    dp[i][j] = dp[i-1][j-1] + 1;
                } else {
                    dp[i][j] = max(dp[i-1][j], dp[i][j-1]);
                }
            }
        }
        return dp[N1][N2];
    }
};